### DISCUSSION TITLE
Unit Tests needed for Godot Motion Matching to combine multiple Godot advanced features correctly

### DISCUSSION THREAD

**Discussion by GeorgeS2019** [GuilhermeGSousa/godot-motion-matching#26] [Ideas]
This project has the potential to unit and integrating test if multiple advanced godot 4.4.x features could be correctly combined together.

The project needs to provide unit and integrating tests ( if possible, in GDscript and c#) to check if certain features are correctly implemented and that such new features implemented do not cause regression, as part of CI/CD delivery.

In time, all the advanced features that need to be combined into this project (partially listed here https://github.com/users/GuilhermeGSousa/projects/2) will be more explicitly defined.

**Comment by GeorgeS2019** [comment]
Working 8th Sept 2024 version: new UI

Sorry for the poor gif, not easy to do it right

![Godot_v4 3-stable_mono_win64_gSdUQA0IEq](https://github.com/user-attachments/assets/093363f6-0246-4889-8970-4a0b4e55957b)

**Reply by GuilhermeGSousa** [reply]
I'm glad you could try it out!

**Comment by GuilhermeGSousa** [comment]
I agree that seeting up testing would be good to do at some point, especially as the API stabilizes a bit. As for 4.4, do you know of features that could be useful for this project? The addition of `SkeletonModifier` in 4.3 was pretty nice, but I'm not in the loop for new features that might be in the next release

**Reply by GeorgeS2019** [reply]
@Lyuma
@Malcolmnixon

=> @GuilhermeGSousa  needs feedback what else for SkeletonModifier in Godot 4.4 could be relevant for Motion Matching
